src/routers/router.ts

```diff
@@ -0,0 +1,202 @@
+                    database.webPushNotificationCollection.findOne({ _id: account.webPushNotification }, (err: Error, wpn: any) => {
+                      if (err) {
+                        console.error("Error fetching webpush notification parameters for account " + account.uniqueId + ' ' + err.message)
+                        res.status(500).send()
+                      } else {
+                        TrackerUtils.watchWebPush(networkId, publicAddress, getWebPushSubstriptionObject(wpn))
+                        res.status(201).json(pub[0]).send()
+                      }
+                    })
+                  } else {
+                    res.status(500).send({ message: 'No web push parameters found for this account' })
+                  }
+                }
+              })
+            }
+          })
+        } else {
+          console.error('Error parsing public address subscription parameters.')
+          res.status(400).send()
+        }
+      } else {
+        res.status(404).send()
+      }
+    }
+  })
+})
+
+// Create WebPushNotification
+// /accounts/:accountId/webpush-notifications
+webpushNotificationRouter.post('/', (req: any, res: any) => {
+  let accountId = req.params.accountId
+  database.accountCollection.findOne({ uniqueId: accountId }, (err: Error, account: any) => {
+    if (err) {
+      console.error("Error occured fetching account " + accountId + ' ' + err.message)
+      res.status(500).send()
+    }
+    if (account) {
+      let webPushNotif
+      try {
+        webPushNotif = {
+          endpoint: req.body.subscription.endpoint,
+          p256dh: req.body.subscription.keys.p256dh,
+          auth: req.body.subscription.keys.auth
+        }
+      } catch (e) {
+        console.error('Error parsing parameters. ' + e)
+        res.status(400).send()
+      }
+      if (webPushNotif) {
+        database.webPushNotificationCollection.insert(webPushNotif, (errWPN: Error, wpn: any) => {
+          if (errWPN) {
+            console.error("Error saving webpush notification. " + errWPN.message)
+            res.status(500)
+          } else {
+            database.accountCollection.update({ _id: account._id }, { $set: { webPushNotification: wpn[0]._id } }, (errUPT: Error) => {
+              if (errUPT) {
+                console.error("Error setting webpush notification on account " + accountId + ' ' + errUPT.message)
+                res.status(500).send()
+              } else {
+                console.info("WebPushNotifictaion set on account " + accountId)
+                res.status(201).json(wpn[0]).send()
+              }
+            })
+          }
+        })
+      }
+    } else {
+      res.status(404).send()
+    }
+  })
+})
+
+// Get all networks
+//networkRouter.get('/', (req: any, res: any) => {})
+
+// Get all enabled medias for an account
+//enabledNotificationMediaRouter.get('/', (req: any, res: any) => {})
+
+let getWebPushSubstriptionObject = (webPushNotificationParameters: any) => {
```

Comment:
Misspelling

Reply:
mappWebPushSubscriptionObject ?

Reply:
`getWebPushSubstriptionObject` => `getWebPushSubscriptionObject`

Reply:
:D You don't like Substription ? :smile: